lib/cldr/helpers/map.ex

```diff
@@ -253,20 +253,20 @@ defmodule Cldr.Map do
   @doc """
   Convert map keys from `camelCase` to `snake_case`
 
-  * `map` is any `Map.t`
+  * `map` is any `t:map/0`
 
   ## Examples
 
   """
-  def underscore_keys(map = %{}) do
+  def underscore_keys(map) when is_map(map) or is_nil(map) do
```

Comment:
I would be curious if that is really called with nil values and if that's not an error by the caller.

Reply:
Yes, other code in the project calls it with potential `nil`s.

Reply:
The potential `nil`s come from the pipelines where the output of `get_in` calls is `|>` to `Cldr.Map.underscore_keys`.  `get_in` could return `nil` if the key does not exist and the calling code doesn't prove it is not `nil`, so dialyzer has to assume it could be `nil`.  However, making `deep_merge` not accept `nil` *may* make dialyzer assume that `get_in` isn't expected to return `nil`, so doing the fix for `deep_merge` so it only accepts `map` could also fix this so it doesn't need to accept `nil`.